Typescript build error

- **Version**:
0.35.6

- **Platform**:
Mac OS X

- **Subsystem**:

#### Severity:
High - Lodestar could not compile

#### Description:
Compilation error when using v0.35.6 with lodestar:

```
@chainsafe/lodestar: $ tsc -p tsconfig.build.json
@chainsafe/lodestar: ../../node_modules/libp2p/dist/src/dialer/index.d.ts:168:18 - error TS2304: Cannot find name 'DialRequest'.
@chainsafe/lodestar: 168     dialRequest: DialRequest;
@chainsafe/lodestar:                      ~~~~~~~~~~~
@chainsafe/lodestar: ../../node_modules/libp2p/dist/src/dialer/index.d.ts:169:17 - error TS2304: Cannot find name 'TimeoutController'.
@chainsafe/lodestar: 169     controller: TimeoutController;
@chainsafe/lodestar:                     ~~~~~~~~~~~~~~~~~
@chainsafe/lodestar: ../../node_modules/libp2p/dist/src/index.d.ts:413:37 - error TS2503: Cannot find namespace 'PersistentPeerStore'.
@chainsafe/lodestar: 413     peerStore?: (PeerStoreOptions & PersistentPeerStore.PersistentPeerStoreOptions) | undefined;
@chainsafe/lodestar:                                         ~~~~~~~~~~~~~~~~~~~
@chainsafe/lodestar: ../../node_modules/libp2p/dist/src/metrics/stats.d.ts:18:23 - error TS2304: Cannot find name 'Big'.
@chainsafe/lodestar: 18         dataReceived: Big;
@chainsafe/lodestar:                          ~~~
@chainsafe/lodestar: ../../node_modules/libp2p/dist/src/metrics/stats.d.ts:19:19 - error TS2304: Cannot find name 'Big'.
@chainsafe/lodestar: 19         dataSent: Big;
@chainsafe/lodestar:                      ~~~
@chainsafe/lodestar: ../../node_modules/libp2p/dist/src/metrics/stats.d.ts:60:23 - error TS2304: Cannot find name 'Big'.
@chainsafe/lodestar: 60         dataReceived: Big;
@chainsafe/lodestar:                          ~~~
@chainsafe/lodestar: ../../node_modules/libp2p/dist/src/metrics/stats.d.ts:61:19 - error TS2304: Cannot find name 'Big'.
@chainsafe/lodestar: 61         dataSent: Big;
@chainsafe/lodestar:                      ~~~
@chainsafe/lodestar: src/chain/chain.ts:85:1 - error TS7008: Member '1' implicitly has an 'any' type.
@chainsafe/lodestar: 85 1
@chainsafe/lodestar:    ~
```

#### Steps to reproduce the error:
We can see the same issue even in the `dist/src` folder of libp2p

The 2nd error is related to the `native-abort-controller` package

```
@chainsafe/lodestar: ../../node_modules/libp2p/node_modules/native-abort-controller/dist/src/index.d.ts:1:10 - error TS2661: Cannot export 'AbortSignal'. Only local declarations can be exported from a module.
@chainsafe/lodestar: 1 export { AbortSignal, AbortController };
@chainsafe/lodestar:            ~~~~~~~~~~~
@chainsafe/lodestar: ../../node_modules/libp2p/node_modules/native-abort-controller/dist/src/index.d.ts:1:23 - error TS2661: Cannot export 'AbortController'. Only local declarations can be exported from a module.
@chainsafe/lodestar: 1 export { AbortSignal, AbortController };
```

--- a/src/dialer/index.js
+++ b/src/dialer/index.js
@@ -56,8 +56,8 @@ const METRICS_PENDING_DIAL_TARGETS = 'pending-dial-targets'
  * @property {Multiaddr[]} addrs
  *
  * @typedef PendingDial
- * @property {DialRequest} dialRequest
- * @property {TimeoutController} controller
+ * @property {import('./dial-request')} dialRequest
+ * @property {import('timeout-abort-controller').TimeoutController} controller
  * @property {Promise<Connection>} promise
  * @property {function():void} destroy
  */

--- a/src/index.js
+++ b/src/index.js
@@ -48,6 +48,8 @@ const { updateSelfPeerRecord } = require('./record/utils')
  * @typedef {import('libp2p-interfaces/src/pubsub').PubsubOptions} PubsubOptions
  * @typedef {import('interface-datastore').Datastore} Datastore
  * @typedef {import('./pnet')} Protector
+ * @typedef {Object} PersistentPeerStoreOptions
+ * @property {number} [threshold]
  */
 
 /**
@@ -110,7 +112,7 @@ const { updateSelfPeerRecord } = require('./record/utils')
  * @property {KeychainOptions & import('./keychain/index').KeychainOptions} [keychain]
  * @property {MetricsOptions & import('./metrics').MetricsOptions} [metrics]
  * @property {import('./peer-routing').PeerRoutingOptions} [peerRouting]
- * @property {PeerStoreOptions & import('./peer-store/persistent').PersistentPeerStoreOptions} [peerStore]
+ * @property {PeerStoreOptions & PersistentPeerStoreOptions} [peerStore]
  * @property {import('./transport-manager').TransportManagerOptions} [transportManager]
  * @property {Libp2pConfig} [config]
  *

--- a/src/metrics/stats.js
+++ b/src/metrics/stats.js
@@ -8,6 +8,7 @@ const retimer = require('retimer')
 
 /**
  * @typedef {import('@vascosantos/moving-average').IMovingAverage} IMovingAverage
+ * @typedef {import('bignumber.js').BigNumber} Big
  */
 
 class Stats extends EventEmitter {

